### PR TITLE
shell.nix: supply `wrapFactor` helper to make standalone factor binary

### PR DESCRIPTION
**Pull request by timor**
This adds the shell function `wrapFactor`.  This function is intended to wrap
the result of calling `build.sh` in the shell environment so it can be executed
outside of the nix shell.

Example:

```
$ nix-shell
[nix-shell] $ ./build.sh bootstrap
...build factor vm and image...
[nix-shell] $ wrapFactor .
exit
$ ./factor
```

`wrapFactor` takes the path to the factor root dir as argument, and expects the
binary `factor` and the image file `factor.image` there and uses Nixpkgs'
`makeWrapper` to wrap the `factor` executable in-place with the correct
`LD_LIBRARY_PATH`.  Afterwards, the factor executable can be called outside of
the nix-shell environment.

### CODE REVIEW

**Comment by timor** [comment]
cc @kwshi @spacefrogg

File: shell.nix [hunk 1]
```diff
@@ -15,13 +15,25 @@ let
...
+    makeWrapper
   ];
+  shellHook = ''
+    wrapFactor () {
+    [ -n "$1" ] || { printf "Usage: wrapFactor <factor-root>" ; return; }
+    local root="$(realpath $1)"
+    local binary="''${root}/factor"
+    wrapProgram "$binary" --prefix LD_LIBRARY_PATH : ${runtimeLibPath} \
+      --argv0 factor
+    ln -sf "''${root}/factor.image" "''${root}/.factor-wrapped.image"
```
Comment (mrjbq7) [inline shell.nix:36]: What's this symlink for?

Comment (spacefrogg) [reply shell.nix:36]: As far as I know, the factor binary searches for an image with the same basename next to it. `wrapProgram` moves the original `$binary` to `.${binary}-wrapped` and establishes a wrapper with the name `$binary`. Thus, I expect the wrapped factor binary to search for a wrapped-named factor image.

I'm just supplying the intent of this code. I don't know if any of my assumptions are correct.

**Comment by mrjbq7** [comment]
Thanks!

File: shell.nix (hunk 1 above)
Comment (timor) [reply shell.nix:36]: Yes, that is the reason for the symlink.

**Pull request merged**